### PR TITLE
Fix failures from many pulls from hub.docker.com (Done)

### LINKED ISSUE #1380: docker image tests failed due to dockerhub limitation
cron-docker-test couldn't pull from dockerhub:
```bash
  /usr/bin/docker pull projectmonai/monai:latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 4.538 seconds before retry.
  /usr/bin/docker pull projectmonai/monai:latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Warning: Docker pull failed with exit code 1, back off 3.538 seconds before retry.
  /usr/bin/docker pull projectmonai/monai:latest
  Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
  Error: Docker pull failed with exit code 1
```

https://github.com/Project-MONAI/MONAI/runs/1567795300?check_suite_focus=true#step:2:14

cc @IsaacYangSLA @YuanTingHsieh @behxyz @yanchengnv

### PR DESCRIPTION
**Pull request by danielschulz**
Fixes #1380 .

### Description
As Docker Hub is applying a rate limit for unauthorized pulls, we need to check how to pull images from it less often. Else, there will be regular failures in the CI pipeline just for technical reasons.

### Status
Work in progress

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [X] small changes like altered copyright year
- [X] changed curly brackets from "$VAR" to "${VAR}"

### CODE REVIEW

**Comment by wyli** [comment]
thanks, this also addresses https://github.com/Project-MONAI/MONAI/issues/1276 cc @IsaacYangSLA

File: Dockerfile [hunk 1]
```diff
@@ -29,7 +31,11 @@ RUN BUILD_MONAI=1 FORCE_CUDA=1 python setup.py develop \
 
 # NGC Client
 WORKDIR /opt/tools
-RUN wget -q https://ngc.nvidia.com/downloads/ngccli_cat_linux.zip && \
+ARG NGC_CLI_SHA256_HASH="03c86402449469a6d12ac2fbbf5c8b460bdbed4f1e01f692e07eb1c144f6de0f"
```
Comment (wyli) [inline Dockerfile:34]: the file will be updated from time to time, please follow https://ngc.nvidia.com/setup/installers/cli for the check

Comment (danielschulz) [reply Dockerfile:34]: Sure, there will be changes. It is good practice however, to check the integrity of downloads. What may I do here? May I version the download so the download and SHA256 hash need to be changed accordingly? Or may I forfeit checking this download entirely?

Comment (wyli) [reply Dockerfile:34]: the link I pasted above has a command for md5 check

Comment (danielschulz) [reply Dockerfile:34]: Thanks. This looks good.

File: Dockerfile [hunk 2]
```diff
@@ -13,8 +13,10 @@ ARG PYTORCH_IMAGE=nvcr.io/nvidia/pytorch:20.10-py3
 
 FROM ${PYTORCH_IMAGE} as base
 
+MAINTAINER Wenqi Li <wenqil@nvidia.com>, Nic Ma <nma@nvidia.com>, Daniel Schulz <danielschulz2005@hotmail.com>
```
Comment (wyli) [inline Dockerfile:16]: this could be consistent with https://github.com/Project-MONAI/MONAI/blob/115f146e4e37e23c7241ac5decfec42bb7aa6c84/setup.cfg#L2-L10

**Comment by danielschulz** [comment]
That's basically it. I'd appreciate you comment on what might be left to do or revise. Thank you.

I am not sure, whether I am able to test all GitHub Actions from my fork. May I just remove the date changes to "2020 - 2021" again or may I leave it in my PR? Thanks for getting back to me.

File: Dockerfile [hunk 3]
```diff
@@ -23,13 +25,21 @@ RUN cp /tmp/requirements.txt /tmp/req.bak \
   && python -m pip install --no-cache-dir --use-feature=2020-resolver -r /tmp/requirements-dev.txt
 
 # compile ext and remove temp files
+# TODO: remark for issue [revise the dockerfile #1276](https://github.com/Project-MONAI/MONAI/issues/1276)
```
Comment (danielschulz) [inline Dockerfile:28]: TODO: remark for issue [revise the dockerfile #1276](https://github.com/Project-MONAI/MONAI/issues/1276)

**Review by wyli — COMMENTED** [review]
thanks, it looks great,  I just have a minor comment about the meta data

**Review by danielschulz — COMMENTED** [review]
> thanks, it looks great, I just have a minor comment about the meta data

I changed it accordingly...

**Review by wyli — APPROVED** [review]
thanks

**Comment by danielschulz** [comment]
> thanks

You're welcome. It was a pleasure to contribute my bit. 

PS: Also your products are great! I am using your GPUs professionally and personally for years now. Really great work from Nvidia! Thank you for this.

**Comment by danielschulz** [comment]
# :tada:

**Pull request merged**